Model/Sync/Catalog/Data.php

```diff
@@ -374,43 +383,42 @@ public function attributeMapping(Product $item)
 
         foreach ($mapAttributes as $key => $attr) {
             try {
-                if ($key === 'gtins') {
-                    $value = $this->prepareGtinsData($attr, $item);
-                } elseif ($key === 'custom_properties') {
-                    $value = $this->prepareCustomProperties($attr, $item);
-                } elseif ($key === 'is_discontinued') {
-                    $value = false;
-                } else {
-                    if ($attr['default']) {
-                        $data = $item->getData($attr['attr_code']);
-
-                        if (isset($attr['type']) && $attr['type'] === 'url') {
-                            $data = $item->getProductUrl();
+                switch ($key) {
+                    case 'gtins':
+                        $value = $this->prepareGtinsData($attr, $item);
+                        break;
+                    case 'custom_properties':
+                        $value = $this->prepareCustomProperties($attr, $item);
+                        break;
+                    case 'is_discontinued':
+                        $value = false;
+                        break;
+                    case 'url':
+                        $value = $item->getProductUrl();
+                        break;
+                    case 'image_url':
+                        $value = $this->getProductImageUrl($item);
+                        break;
+                    default:
+                        if (!$attr['default'] && isset($attr['method']) && $attr['method']) {
+
+                            $configKey = isset($attr['attr_code']) && $attr['attr_code'] ?
+                                $attr['attr_code'] : '';
+
+                            $method = $attr['method'];
+                            $itemValue = $this->$method($item, $configKey);
+                            $value = $itemValue ?: ($method == 'getProductPrice' ? 0.00 : $itemValue);
+                        } else {
+                            $value = '';
                         }
-
-                        if (isset($attr['type']) && $attr['type'] === 'image') {
-                            $baseUrl = $this->yotpoCoreConfig->getBaseUrl(UrlInterface::URL_TYPE_MEDIA);
-                            $data = $data ? $baseUrl . 'catalog/product' . $data : "";
+                        if ($key == 'group_name' && $value) {
+                            $value = strtolower($value);
+                            $value = str_replace(' ', '_', $value);
+                            $value = preg_replace('/[^A-Za-z0-9_-]/', '-', $value);
```

Comment:
What does this line do? It seems like everything regex search

Reply:
It replaces everything that is **not** alphanumeric, underscore, and dash to dash.

Reply:
Thanks for the explanation. Is this the expected behaviour? do we want it to replace !@#$%^&*()+= to dash?